crates/core/tedge_mapper/src/operations.rs

```diff
@@ -1,118 +1,138 @@
 use std::{
-    collections::{HashMap, HashSet},
+    collections::HashMap,
     fs,
     path::{Path, PathBuf},
 };
 
+use serde::Deserialize;
+
 use crate::error::OperationsError;
 
 /// Operations are derived by reading files subdirectories per cloud /etc/tedge/operations directory
 /// Each operation is a file name in one of the subdirectories
 /// The file name is the operation name
 
-type Cloud = String;
-type OperationName = String;
-type Operation = HashSet<OperationName>;
-type OperationsMap = HashMap<Cloud, Operation>;
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub struct Exec {
```

Comment:
The name `Exec` is not really explicit. Would `OnMessageExec` be better ? `OnMessageCommand` ?